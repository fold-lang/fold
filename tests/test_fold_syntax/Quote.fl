// Code quoting
x = quote a;
x = quote 1;
x = quote ("abc");
x = quote (f x);
x = quote (1 + 2);
x = quote [1, 2, 3];
x = quote ((42));
x = quote (let x = 1);
x = quote (let a = 42; a + 1);

// Code antiquoting
int_code = quote 42;
x = quote (1000 + unquote int_code);
x = quote (2000 + unquote (quote 42))

