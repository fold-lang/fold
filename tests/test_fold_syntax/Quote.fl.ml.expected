let x = Ident (Lower "a")
let x = Const (Int 1)
let x = Const (String "abc")
let x = Seq (None, [Ident (Lower "f"); Ident (Lower "x")])
let x = Seq (None, [Sym "+"; Const (Int 1); Const (Int 2)])
let x =
  Scope
    ("[", (Seq ((Some ","), [Const (Int 1); Const (Int 2); Const (Int 3)])),
      "]")
let x = Const (Int 42)
let x = Shape ("let", [Shape ("=", [Ident (Lower "x"); Const (Int 1)])])
let x =
  Seq
    ((Some ";"),
      [Shape ("let", [Shape ("=", [Ident (Lower "a"); Const (Int 42)])]);
      Seq (None, [Sym "+"; Ident (Lower "a"); Const (Int 1)])])
let int_code = Const (Int 42)
let x = Seq (None, [Sym "+"; Const (Int 1000); int_code])
let x = Seq (None, [Sym "+"; Const (Int 2000); Const (Int 42)])
